src/main/main.js

```diff
@@ -7,19 +7,68 @@ process.env.PATH += ':/usr/local/bin';
 var mainWindow;
 
 app.on('ready', function () {
+	mainWindow = createWindow();
+});
 
-    var atomScreen = require('screen');
-    var size = atomScreen.getPrimaryDisplay().workAreaSize;
-    mainWindow = new BrowserWindow({
-        width: size.width,
-        height: size.height,
-	'web-preferences': {
-		'overlay-scrollbars': true
-	},
-        resizable: true
-    });
+app.on('window-all-closed', function() {
+	if(process.platform != 'darwin') {
+		app.quit();
+	}
+});
 
-    mainWindow.loadUrl('file://' + __dirname + '/../../index.html');
-    mainWindow.focus();
-    menu.setMenu(app, mainWindow);
+app.on('activate-with-no-open-windows', function () {
+	mainWindow = createWindow();
 });
+
+function createWindow() {
+	var path = require("path");
+	var fs = require("fs");
+	var userPrefsPath = path.join(app.getDataPath(), "preferences.json");
```

Comment:
We already have serialization setup. Could you try to use it? https://github.com/black-screen/black-screen/blob/master/src/Terminal.ts#L98